airflow/www/views.py

```diff
@@ -3574,13 +3586,27 @@ def datasets_summary(self):
                     DatasetModel.id,
                     DatasetModel.uri,
                 )
-                .filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
                 .order_by(*order_by)
-                .offset(offset)
-                .limit(limit)
-                .all()
-            ]
-            data = {"datasets": datasets, "total_entries": total_entries}
+            )
+
+            if updated_before or updated_after:
+                count_query = count_query.outerjoin(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id)
```

Comment:
are you sure you want outerjoin?

Reply:
Isn't that what you suggested?

Reply:
For the unfiltered query we want outer joins.

But for the filtered results query and for the filtered count query, doing an outer join and then filtering the results later in a WHERE clause is effectively an inner join, right?

I (possibly naively!) would expect that most query planners will properly optimize that into an inner join, so the performance difference should be nothing or negligible. And performance is always such a fickle thing to preemptively plan for and measure after the fact.

Part of me thinks that we keep it as similar to the results query as possible, and use outer joins and filter it later on.

Thoughts?

Reply:
> Isn't that what you suggested?

I just pointed out that it was sort of ambiguous what the intention since using outer join in combination with those filters effectively makes it an inner join.

My the question I'm highlighting is about logic not necessarily performance.  The question is, if a dataset has no events in the time range, do you still want it to appear in the result?  If that's true you need to do an outer join but move the filters to join condition.

I can hop on a call if nec.

Reply:
We only want datasets with an event in the time range.